core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/negotiation/ConsumerContractNegotiationManagerImpl.java

```diff
@@ -298,22 +303,23 @@ private int sendContractOffers() {
      * @return the number of processed ContractNegotiations.
      */
     private int sendCounterOffers() {
-        var processes = negotiationStore.nextForState(ContractNegotiationStates.CONSUMER_OFFERING.code(), batchSize);
+        var processes = negotiationStore.nextForState(CONSUMER_OFFERING.code(), batchSize);
 
         for (ContractNegotiation process : processes) {
             var offer = process.getLastContractOffer();
-            var response = sendOffer(offer, process, ContractOfferRequest.Type.COUNTER_OFFER);
-            if (response.isCompletedExceptionally()) {
-                process.transitionOffering();
-                monitor.debug(format("[Consumer] Failed to send contract offer with id %s. ContractNegotiation %s stays in state %s.",
-                        offer.getId(), process.getId(), ContractNegotiationStates.from(process.getState())));
-                continue;
-            }
-
-            process.transitionOffered();
-            monitor.debug(String.format("[Consumer] ContractNegotiation %s is now in state %s.",
-                    process.getId(), ContractNegotiationStates.from(process.getState())));
-            negotiationStore.save(process);
+            sendOffer(offer, process, ContractOfferRequest.Type.COUNTER_OFFER)
```

Comment:
Thanks a lot for this PR! Could we also take opportunity of this change to display the throwable message within the logs? Otherwise when sendOffer fails then you do not get any information why.